src/movement_controls.py

```diff
@@ -0,0 +1,316 @@
+        elif input.y_axis.value < -0.1 and input.ry_axis.value < -0.1:
+            acceleration_change -= self.acceleration_rate
+
+        # the current acceleration is used to have speed progress from zero to its maximum
+        self.current_acceleration = self.clamp(
+            self.current_acceleration + (acceleration_change * delta_time), -1, 1
+        )
+
+        car_angle = math.radians(vehicle.angle)
+
+        # find the X and Y movement based on the angle of the sprite
+        self.current_velocity_x = (
+            self.drive_speed
+            * math.cos(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+        self.current_velocity_y = (
+            self.drive_speed
+            * math.sin(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+
+    def ghost(self, delta_time, input, vehicle):
+        # move in the direction of the x/y axis input relative to the car's angle
+
+        acceleration_change = 0
+        if input.brake_axis.value:
+            acceleration_change -= self.decceleration_rate
+        elif input.x_axis.value != 0 or input.y_axis.value != 0:
+            acceleration_change += self.acceleration_rate
+        else:
+            if abs(self.current_acceleration) != 0:
+                if abs(self.current_acceleration) <= delta_time * self.friction:
+                    self.current_acceleration = 0
+                else:
+                    acceleration_change -= delta_time
+
+        # the current acceleration is used to have speed progress from zero to its maximum
+        self.current_acceleration = self.clamp(
+            self.current_acceleration + (acceleration_change * delta_time), 0, 1
+        )
+
+        car_angle = math.radians(vehicle.angle)
+
+        xNormalized = input.x_axis.value
+        yNormalized = input.y_axis.value
+
+        if abs(input.y_axis.value) + abs(input.x_axis.value) > 1:
+            xNormalized = input.x_axis.value / (
+                abs(input.y_axis.value) + abs(input.x_axis.value)
+            )
+            yNormalized = input.y_axis.value / (
+                abs(input.y_axis.value) + abs(input.x_axis.value)
+            )
+
+        self.current_velocity_x = (
+            self.drive_speed
+            * (
+                (math.cos(car_angle) * input.y_axis.value)
+                + (math.sin(car_angle) * input.x_axis.value)
+            )
+            * self.current_acceleration
+            * delta_time
+        )
+        self.current_velocity_y = (
+            self.drive_speed
+            * (
+                (math.sin(car_angle) * input.y_axis.value)
+                - (math.cos(car_angle) * input.x_axis.value)
+            )
+            * self.current_acceleration
+            * delta_time
+        )
+
+        # a stationary car cant turn
+        self.current_turnVelocity = input.rx_axis.value * self.turn_speed * delta_time
```

Comment:
This turns backwards because a positive angle is counterclockwise while a positing rx_axis is to the right. Just needs to be
-input.rx_axis.value